### PR TITLE
add port for GCC/MSP430FR5969

### PR DESCRIPTION
**Pull request by rphh**
Port for MSP430X and GCC.

Compiler: msp430-elf-gcc (Mitto Systems Limited - msp430-gcc 8.3.0.16) 8.3.0
Board: MSP430FR5969LaunchPad
Support for small memory model, and large memory model

### CODE REVIEW

File: portable/GCC/MSP430FR5969/portmacro.h [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+}
+/*-----------------------------------------------------------*/
+
+/* Task utilities. */
+extern void vPortYield( void ) __attribute__ ( ( naked ) );
+#define portYIELD()         vPortYield()
+#define portNOP()           asm volatile ( "NOP" )
+/*-----------------------------------------------------------*/
+
+/* Hardwware specifics. */
```
Comment (dcgaws) [inline portable/GCC/MSP430FR5969/portmacro.h:119]: "Hardware"

File: portable/GCC/MSP430FR5969/port.c [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+        pxTopOfStack--;
+        *pxTopOfStack = ( StackType_t ) 0xbbbb;
+        pxTopOfStack--;
+        *pxTopOfStack = ( StackType_t ) 0xaaaa;
+        pxTopOfStack--;
+        *pxTopOfStack = ( StackType_t ) 0x9999;
+        pxTopOfStack--;
+        *pxTopOfStack = ( StackType_t ) 0x8888;
+        pxTopOfStack--;
+        *pxTopOfStack = ( StackType_t ) 0x777;
```
Comment (dcgaws) [inline portable/GCC/MSP430FR5969/port.c:203]: Should be 0x7777?

File: portable/GCC/MSP430FR5969/port.c [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+#endif
+
+/*
+ * Macro to restore a task context from the task stack.  This is effectively
+ * the reverse of portSAVE_CONTEXT().  First the stack pointer value is
+ * loaded from the task control block.  Next the value for usCriticalNesting
+ * used by the task is retrieved from the stack - followed by the value of all
+ * the general purpose msp430 registers.
+ *
+ * The bic instruction ensures there are no low power bits set in the status
```
Comment (dcgaws) [inline portable/GCC/MSP430FR5969/port.c:100]: Looks like the bic instruction has been removed in this port.

Comment (rphh) [reply portable/GCC/MSP430FR5969/port.c]: I removed this comment. These bits are cleared in `prvTickISR` using `__bic_SR_register_on_exit`.

**Comment by rphh** [comment]
@cobusve @dcgaws Is there something else what I can do?

**Comment by aggarg** [comment]
Moved this to community supported ports repo - https://github.com/FreeRTOS/FreeRTOS-Kernel-Community-Supported-Ports/pull/4

**Pull request closed without merging**